packages/core/src/components/popover/popover.tsx

```diff
@@ -336,22 +322,18 @@ export class Popover extends AbstractPureComponent<IPopoverProps, IPopoverState>
     public componentWillReceiveProps(nextProps: IPopoverProps) {
         super.componentWillReceiveProps(nextProps);
 
-        const nextDisabled = getDisabled(nextProps);
+        const nextIsOpen = this.getIsOpen(nextProps);
 
-        if (nextProps.isOpen == null && nextDisabled && !this.state.disabled) {
-            // ok to use setOpenState here because disabled and isOpen are mutex.
+        if (nextProps.isOpen != null && nextIsOpen !== this.state.isOpen) {
+            this.setOpenState(nextIsOpen);
+            if (nextProps.isOpen != null) {
```

Comment:
this condition seems tautological -- it's already checked on line 327

Reply:
Ah yep, I think I had a different check here before. Will remove.